pkg/multiTenancyPlugins/naming/nameScoping.go

```diff
@@ -118,3 +129,147 @@ func (nameScoping *DefaultNameScopingImpl) Handle(command string, cluster cluste
+	
+	// create affinity array
+	affinityRef := make([]string,0)
+	env := containerConfig.Env
+	var affinityIndex int
+	affinityLabelFirst := false
+	for i, e := range env {
+		if strings.HasPrefix(e,"affinity:") {
+		  if strings.HasPrefix(e,"affinity:image==") {
+			break  // ignore affinity for images 
+		  } else if strings.HasPrefix(e,"affinity:container==") {
+			containerRefIndex := strings.Index(e,"affinity:container==")+len("affinity:container==")
+			affinityRef = append(affinityRef,e[containerRefIndex:len(e)])
+			affinityIndex = i
+			break
+		  } else { // affinity:<label>:<value>
+			labelRefIndex := strings.Index(e,"affinity:")+len("affinity:")
+			affinityRef = append(affinityRef,e[labelRefIndex:len(e)])
+			affinityIndex = i
+			affinityLabelFirst = true
+			break
+		  }
+		}
+	}
+	
+	// create links array 
+	links := make([]string,0) 
+	for _, link := range containerConfig.HostConfig.Links {
+		linkSplit := strings.SplitN(link,":",2)
+		links = append(links,strings.TrimSpace(linkSplit[0]))
+	}
+	
+	var m map[string]string
+	var err error
+	containerRefs := make([]string,0)
+	containerRefs = append(containerRefs,affinityRef...)  
+	containerRefs = append(containerRefs,volumesFrom...)  
+	containerRefs = append(containerRefs,links...)		
+	if m,err = getIDsFromContainerReferences(cluster, tenantId, containerRefs,affinityLabelFirst); err != nil {
+		return err
+	}
+	// update containerConfig
+	// update VolumesFrom
+	for i,k := range containerConfig.HostConfig.VolumesFrom {
+		containerConfig.HostConfig.VolumesFrom[i] = m[k]			
+	}
+	// update affinity
+	if len(affinityRef) > 0 {
+		containerConfig.Env[affinityIndex] = "affinity:container==" + m[affinityRef[0]]
+	}
+	// update links
+	links = make([]string,0)
+	linkSet := make(map[string]bool) 
+	for _, link := range containerConfig.HostConfig.Links {
+		linkSplit := strings.SplitN(link,":",2)
+		containerIdName := strings.TrimSpace(linkSplit[0])
+		containerId := m[containerIdName]
+		if containerId != containerIdName {
+			linkSet[containerId + ":" + containerIdName] = true
+		}
+		if len(linkSplit) > 1 {
+			linkSet[containerId + ":" + strings.TrimSpace(linkSplit[1])] = true
+		}
+		for k,_ := range linkSet {
+			links = append(links,k)
+		}		
+	}
+	containerConfig.HostConfig.Links = links
+	return nil
+
+}
+func getIDsFromContainerReferences(cluster cluster.Cluster, tenantId string, containerReferences[] string, affinityLabelFirst bool ) (map[string]string, error) {
+	// containerReferences is a array of container references of the form long id, short id, or name
+	// if affinityLabelFirst is true the first element of containerReferences an affinity label reference of the form label:value
+	m := make(map[string]string)
+	for _, containerReference := range containerReferences {
+		m[containerReference] = ""
+	} 
+	for _, container := range cluster.Containers() {
+		if container.Labels[headers.TenancyLabel] == tenantId {
+		  for i, containerReference := range containerReferences {
```

Comment:
This is not efficient you check again and again even if you already found this specific reference.